### PR TITLE
Update and better manage build environment

### PR DESCRIPTION
**Pull request by grugnog**
Originally looking at enabling collapsible nav, but decided to leave it disabled. This does include a number of other improvements to the RTD/mkdocs configuration though.

### CODE REVIEW

**Review by dmundra — APPROVED** [review]
Works https://handbook.civicactions.com/en/collapsiblenav/050-how-we-work/practice-areas/!

But has the same issues as the current navigation (the menu doesn't stay active, and bounces back to the top).

File: mkdocs.yml [hunk 1]
```diff
@@ -4,11 +4,10 @@ site_url: https://handbook.civicactions.com/en/latest/
 theme:
   name: readthedocs
   highlightjs: true
+  collapse_navigation: true
```
Comment (dmundra) [inline mkdocs.yml]: Discussed with @grugnog that we want to keep other changes in this PR but revert the `collapse_navigation` change.

**Review by dmundra — CHANGES_REQUESTED** [review]
Left a comment about reverting the collapse navigation change.

**Comment by dmundra** [comment]
@grugnog you disabled the collapsible nav but it still is appearing on the branch site https://handbook.civicactions.com/en/collapsiblenav/000-contributing/docs-governance/. Maybe `collapse_navigation: false` has to be removed completely?

**Review by dmundra — CHANGES_REQUESTED** [review]
Left a comment https://github.com/CivicActions/handbook/pull/912#issuecomment-1320687598

**Comment by dmundra** [comment]
Interesting, even after removing the collapsiblenav and reverting mkdocs.yml completely this version https://handbook.civicactions.com/en/collapsiblenav/ still displays the collapsiblenav feature. Is the branch name causing it (doubt it), is it some extreme caching?

@grugnog thoughts? Maybe new PR for the changes and close this one?

File: .readthedocs.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+
+# Required
+version: 2
+
+# Set the version of Python and other tools you might need
+build:
+  os: ubuntu-20.04
+  tools:
+    python: "3.9"
```
Comment (dmundra) [inline .readthedocs.yaml:12]: Hmm maybe it is the version of python and mkdocs that is the issue.

collapsiblenav build: https://readthedocs.org/projects/civicactions-handbook/builds/18803675/

vs

another PR build: https://readthedocs.org/projects/civicactions-handbook/builds/18782556/

File: requirements.in [hunk 3]
```diff
@@ -0,0 +1,2 @@
+mkdocs==1.4.2
```
Comment (dmundra) [inline requirements.in:1]: Version of mkdocs is definitely buggy causing `collapse_navigation: false` to not work.

**Comment by grugnog** [comment]
After playing with this, it seems like the upgrade changes quite a few things in unexpected ways and rather than figure out which we want to keep/drop etc I ported these changes to https://github.com/CivicActions/handbook/pull/926 which is the direction the design team is looking at going forward.

**Pull request closed without merging**